src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -199,14 +199,16 @@ private ImmutableArray<ClrInfo> GetOrCreateClrVersions()
                         {
                             if (runtimeInfo.RuntimeModuleIndex[0] >= sizeof(int) + sizeof(int))
                             {
-                                runtimeTimeStamp = BitConverter.ToInt32(new ReadOnlySpan<byte>(runtimeInfo.RuntimeModuleIndex + sizeof(byte), sizeof(int)).ToArray(), 0);
-                                runtimeFileSize = BitConverter.ToInt32(new ReadOnlySpan<byte>(runtimeInfo.RuntimeModuleIndex + sizeof(byte) + sizeof(int), sizeof(int)).ToArray(), 0);
+                                runtimeTimeStamp = BitConverter.ToInt32(new ReadOnlySpan<byte>(runtimeInfo.DacModuleIndex + sizeof(byte), sizeof(int)).ToArray(), 0);
+                                runtimeFileSize = BitConverter.ToInt32(new ReadOnlySpan<byte>(runtimeInfo.DacModuleIndex + sizeof(byte) + sizeof(int), sizeof(int)).ToArray(), 0);
+                                dacAgnosticName = ClrInfoProvider.GetDacFileName(flavor, platform);
                             }
                         }
                         else
                         {
                             // The first byte of the module indexes is the length of the build id
                             runtimeBuildId = new ReadOnlySpan<byte>(runtimeInfo.RuntimeModuleIndex + sizeof(byte), runtimeInfo.RuntimeModuleIndex[0]).ToArray().ToImmutableArray();
```

Comment:
Is this supposed to be the runtime's build id or the DAC's?  I'm confused now. It looks like you changed what runtimeTimeStamp/runtimeFileSize from the runtime's (coreclr's) to the DAC's. But for non-single-file apps, it is the timestamp/filesize or build id of coreclr.

When I added the single-file support here I may have misunderstood what the parameters to DacInfo constructor should be.

Reply:
Oddly enough, both are archived!  You can request `mscordaccore.dll` under either CLR or the Dac's own properties and it shows up.  I thought it would be more consistent and reasonable to request under the dac's properties, even if both currently work.

It's really the long name `mscordaccore_*_*_*.dll` which is supposed to be under clr'dll's properties, but we need the version to request that one.

Reply:
Yes, both are archived on purpose under the special clr keys.

Reply:
Even though they indexed under both keys, it seems to me that the definition of DacInfo.IndexFileSize/DacInfo.IndexTimeStamp was coreclr's. It looks like that is true if it is a non-singlefile-app.

Reply:
I guess this is a technicality if those properties are only used to download the DAC.

Reply:
This did break SOS (I'm looking at the DARC update PR right now) because it assumes that DacInfo.IndexFileSize/DacInfo.IndexTimeStamp is the runtime index info. I guess SOS is made an incorrect assumption. Why did you change this again?

Reply:
The dac should be indexed (and requested under these properties:

long - mscordaccore_amd64_amd64_1.2.3.4.dll [clr-timestamp] [clr-build-filesize]
short - mscordaccore.dll [dac-timestamp] [dac-filesize]

Because we are requesting the short named dac "mscordaccore.dll" in this scenario, we should be requesting it with its own properties, right?

Reply:
Both the long and short named DAC is indexed with both the identity key and coreclr's.  SOS and dotnet-symbol has depended on that for a long time. But for the long named DAC it should be the coreclr index for cross-architecture DACs.  

long - same architecture: indexed under both keys
long - cross architecture (i.e. mscordaccore_amd64_arm64_1.2.3.4.dll): indexed only under the coreclr key
short - always indexed under both keys

/cc: @hoyosjs

Reply:
We are talking about single-file scenario here?  Or a regular CLR debugging scenario?  The code in question is ONLY hit for single-file.  And we provide the short name and dac properties.  Take a look:

``` C#
        static void Main(string[] args)
        {
            DataTarget dt = DataTarget.LoadDump(@"D:\work\single_file_full.dmp");
            foreach (ClrInfo info in dt.ClrVersions)
            {
                Console.WriteLine($"CLR: {info.ModuleInfo.FileName} {info.Flavor} {info.Version}");
                Console.WriteLine($"Dac: {info.DacInfo.PlatformSpecificFileName} timestamp:{info.DacInfo.IndexTimeStamp:x} filesize:{info.DacInfo.IndexFileSize}");

                string result = dt.BinaryLocator?.FindBinary(info.DacInfo.PlatformSpecificFileName, info.DacInfo.IndexTimeStamp, info.DacInfo.IndexFileSize, checkProperties: false);

                Console.WriteLine(result);

                // We use this:
                // DataTarget.BinaryLocator?.FindBinary(DacInfo.PlatformSpecificFileName, DacInfo.IndexTimeStamp, DacInfo.IndexFileSize, checkProperties: false);
            }
        }
```

```
CLR: c:\git\clrmd\src\TestTargets\single-file\publish\single-file.exe Core 1.0.0.0
Dac: mscordaccore.dll timestamp:6216ca94 filesize:1331200
SUCCESS: C:\Users\leecu\AppData\Local\Temp\symbols\mscordaccore.dll\6216ca94145000\mscordaccore.dll
```

So as you can see, this is a single file .exe, and we are providing the **short named dac** along with the **dac's filesize and timestamp**, and it's successfully requested from the symbol server.

Can you please run the code above on the dump that's failing?  I'm happy to make changes here if going back to the previous behavior is correct but first we need to understand what exactly is happening.

Reply:
Also, perhaps while making breaking changes, it might be a good idea to redesign this DacInfo property and simply enumerate all valid dacs (which there can be multiple of) but I want to get to the bottom of the failure you are seeing first.

Reply:
I should have been clearer about the SOS failure. It is the clrstack -i command that downloads DBI that is failing. All the other DAC related tests are ok.  The reason for the failure is the assumption I made about the DacInfo: that is is the coreclr index info.  SOS uses the DAC timestamp/file size to build the DBI index which after your change to the actual DAC identity key breaks the download.  So you could correctly argue that SOS is using these values incorrectly.  I can fix SOS to build the DBI key from the ClrRuntimeInfo.DbiModuleIndex.

Reply:
Ahhh ok!  That makes more sense!  Don't update SOS yet.  Let me look at the design here and provide you the right information.  We are already in the realm of breaking changes, so at this point let me build something that's actually the data you want in a sensible place.

Reply:
I'll have something for you by end of day.

Reply:
There is no hurry. You could do all the work in your other breaking PR #995. 

In the end, it should be clear what kind of index info it is (just like I suggested in 995): "identity" or "coreclr".  I am ok with the new assumption about DacInfo index info being the DAC index but it needs to be consistent across single-file and non-single-file apps.  SOS can always get the "coreclr" info from ClrInfo.ModuleInfo.

Reply:
I'm starting to work on this now.

One issue right now is that `ClrInfo.ModuleInfo` in single-file mode just gives you information about the single-file exe.  It's too strange to change that behavior to be CLR's info.  It's best to just expose a new API that gives you all of the relevant information I think.  ClrMD was designed to be a runtime inspection tool, and giving deep information about all of the various pieces of CLR diagnostic artifacts is well within its purview. I've just never sat down to enumerate all of the various bits of information that's available about CLR diagnostic artifacts.  The end result will be cleaner than what we have now I think.

I hope to have something by tomorrow.  The change I intend to make is a bit bigger than just spot fixing this issue.